pkg/disttask/framework/scheduler/scheduler_test.go

```diff
@@ -535,6 +535,9 @@ func TestManagerDispatchLoop(t *testing.T) {
 	scheduler.RegisterSchedulerFactory(proto.TaskTypeExample,
 		func(ctx context.Context, task *proto.Task, param scheduler.Param) scheduler.Scheduler {
 			idx := counter.Load()
+			if int(idx) > len(concurrencies) {
```

Comment:
might panic again when return nil

Reply:
we should add a check in `mockScheduler.EXPECT().ScheduleTask().Do(func() {`, when task already success, just return